subscribe/lambda/function.rb

```diff
@@ -56,7 +58,7 @@ def skip?
       def subscribe!
         !!cloudwatch.put_subscription_filter(
           log_group_name: cloudwatch_log_group_name,
-          filter_name: 'ShipToLoggly',
+          filter_name: FILTER_NAME,
           filter_pattern: filter_pattern,
           destination_arn: destination_arn
```

Comment:
is destination_arn the new version of the ship to loggly lambda?

Reply:
That's correct. 

When updating `cloudwatch_loggly` to a new version, the current process requires deletion of the previous SAM application. When the subscribe/push functions are deleted, the subscription filters to the older lambdas remain as a dangling reference.

Previously, the subscribe function would only check if the `filter_pattern` was stale to determine if a subscription filter needs updating. The new logic checks if the `destination_arn` or the `filter_pattern` are stale to determine if an update is needed.